### PR TITLE
monocle top markers

### PR DESCRIPTION
**Pull request by pcm32**
This PR adds the markers test and markers filtering functionality to produce top markers in monocle. There are a few things that I'm not sure how they work in the current layout of the package, for instance whether I need to do something to inhering the output_plot and output_plot_format, or how to deal with the case where the method can output more than a single table (looking to inherit the output_table option).

@nh3 could you advise on what I'm missing to complete this? Thanks!

### CODE REVIEW

**Comment by pcm32** [comment]
Also, I think that I'm using some dplyr constructs, is there a way to specify a library call for this particular function (I could call it inside the function, but I wonder if you are using some other way). dplyr is part of the conda package for monocle3-cli.

**Comment by pcm32** [comment]
I have removed as much as possible redundant travis.org and travis.com (to leave only the last one for PRs), to reduce our carbon footprint :-).

**Comment by nh3** [comment]
Many thanks for the PR @pcm32! I can do the rest. Just to confirm, you want it to always write a full marker table, an optional top marker table and an optional marker-cluster plot, is that right?

**Comment by pcm32** [comment]
Yes, sounds right, unless that you think that top markers output should be the relevant one and the full one the optional. Both ways work for me. Thanks @nh3! I can do release and so on later if you want.

**Comment by pcm32** [comment]
Hehehe, apparently I was a bit off on how to implement this.

File: R/cmd_methods.R [hunk 1]
```diff
@@ -352,3 +358,71 @@ monocle_plotCells <- function(
...
+
+    marker_test_res <- do.call(
+        'top_markers',
+        c(list(cds, verbose = verbose), topMarkers_options)
+    )
+
+    top_marker_test_res <- marker_test_res %>%
+        filter(fraction_expressing >= filter_fraction_expression) %>%
+        group_by(cell_group) %>%
+        top_n(top_n_markers, pseudo_R2) %>%
```
Comment (nh3) [inline R/cmd_methods.R]: `pseudo_R2` only exists when `--marker-sig-test` is set.
When not set, the `marker_test_res` looks like the following:
```
   gene_id  cell_group marker_score specificity mean_expression fraction_expres…
   <chr>    <chr>             <dbl>       <dbl>           <dbl>            <dbl>
 1 WBGene0… 1                 0.303       0.303           38.9             0.999
 2 WBGene0… 1                 0.397       0.406            6.43            0.976
 3 WBGene0… 1                 0.288       0.393            2.01            0.733
 4 WBGene0… 1                 0.289       0.327            1.95            0.883
 5 WBGene0… 1                 0.291       0.307            8.46            0.948
 6 WBGene0… 1                 0.277       0.292           13.1             0.948
 7 WBGene0… 1                 0.340       0.340           86.3             0.999
 8 WBGene0… 1                 0.514       0.581            5.39            0.884
 9 WBGene0… 1                 0.444       0.492            3.19            0.903
10 WBGene0… 1                 0.469       0.518            2.90            0.905
```

**Comment by nh3** [comment]
The inner workings aren't straightforward and I didn't write documentation, it took me sometime too. In the end, I chose to save top markers as a mandatory output.

I think this is now ready for review.

**Comment by pcm32** [comment]
Being the author of the PR, I cannot review it. @pinin4fjords or @a-solovyev12 could you please take a look? Thanks @nh3 for the changes, it looks good to me at least.

**Review by pinin4fjords — APPROVED** [review]
I don't see obvious issues, it's not yet production code, and I think we can trust @nh3 's good work. Approved.

**Comment by nh3** [comment]
Thank you @pinin4fjords and @pcm32. Is there a way to make travis CI run the other tests? Otherwise it seems it wouldn't allow us to merge without force.

**Comment by pinin4fjords** [comment]
Try pushing a dummy spacing change or something @nh3 ? Has worked in previous situations where Travis got confused.

**Comment by nh3** [comment]
Thanks @pcm32! Actually I wonder how did you do this. Was always a bit tired to see the same tests run 2-3 times...

> I have removed as much as possible redundant travis.org and travis.com (to leave only the last one for PRs), to reduce our carbon footprint :-).

**Comment by pcm32** [comment]
I went to the travis-ci.org and .com dashboard, once I was sure that this was active in the travis-ci.com side, I turned off the travis-ci.org ones. Then I only activated the PR one on the travis-ci.com (turned off the branch one). But we can use branch if you want instead of PR, I just wouldn't use both.

**Pull request merged**